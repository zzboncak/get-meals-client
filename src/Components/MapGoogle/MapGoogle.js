import React, { useState } from "react";
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"


const MapGoogle = withScriptjs(
    withGoogleMap(props => {
        const [selectedLocation, setSelectedLocation] = useState(null)
        return (

            <GoogleMap
                defaultZoom={9}
                defaultCenter={{ lat: 37.0902, lng: -95.7129 }}
                center={{ lat: props.latlonArray[0], lng: props.latlonArray[1] }}
            >

                {props.locations.map(coordinates => (

                    <Marker
                        key={coordinates.id}
                        position={{
                            lat: coordinates.location_latitude,
                            lng: coordinates.location_longitude
                        }}
                        onClick={() => {
                            setSelectedLocation(coordinates);

                        }}

                    />
                ))}

                {selectedLocation && (
                    <InfoWindow
                        position={{
                            lat: selectedLocation.location_latitude,
                            lng: selectedLocation.location_longitude
                        }}
                        onCloseClick={() => {
                            setSelectedLocation(null);
                        }}
                    >
                        <>
                            <h3> {selectedLocation.location_name} </h3>
                            <div> Hours: {selectedLocation.open_hour} - {selectedLocation.close_hour} </div>
                            <div> Address: {`${selectedLocation.street_address} ${selectedLocation.city} ${selectedLocation.state} ${selectedLocation.zip}`} </div>
                            <div> Website: <a href={selectedLocation.website}>{selectedLocation.website}</a> </div>
                            <div> Description: {selectedLocation.location_description} </div>

                        </>
                    </InfoWindow>
                )}
            </GoogleMap>
        )
    })
)

export default MapGoogle