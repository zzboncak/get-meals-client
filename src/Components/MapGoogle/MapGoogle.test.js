import React from 'react';
import ReactDOM from 'react-dom';
import MapGoogle from './MapGoogle';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";

describe(`Map component`, () => {
  const props = {
    locations: [],
  };

  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<MapGoogle 
        googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=
        ${process.env.REACT_APP_GOOGLE_KEY}
        `}
        loadingElement={<div style={{ height: "100%" }} />}
        containerElement={<div style={{ height: "400px" }} />}
        mapElement={<div style={{ height: "100%" }} />}
        {...props} />, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <MapGoogle 
              googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=
              ${process.env.REACT_APP_GOOGLE_KEY}
              `}
              loadingElement={<div style={{ height: "100%" }} />}
              containerElement={<div style={{ height: "400px" }} />}
              mapElement={<div style={{ height: "100%" }} />}
              {...props} />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})