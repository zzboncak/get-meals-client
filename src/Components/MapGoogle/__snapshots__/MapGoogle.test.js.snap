// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders the component as expected 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
